src/Symfony/Component/DependencyInjection/Tests/Compiler/ResolveInvalidReferencesPassTest.php

```diff
@@ -23,17 +23,63 @@ public function testProcess()
         $container = new ContainerBuilder();
         $def = $container
             ->register('foo')
-            ->setArguments(array(new Reference('bar', ContainerInterface::NULL_ON_INVALID_REFERENCE)))
+            ->setArguments(array(
+                new Reference('bar', ContainerInterface::NULL_ON_INVALID_REFERENCE),
+                new Reference('baz', ContainerInterface::IGNORE_ON_INVALID_REFERENCE),
+            ))
             ->addMethodCall('foo', array(new Reference('moo', ContainerInterface::IGNORE_ON_INVALID_REFERENCE)))
         ;
 
         $this->process($container);
 
         $arguments = $def->getArguments();
-        $this->assertNull($arguments[0]);
+        $this->assertSame(array(0 => null, 1 => null), $arguments);
         $this->assertCount(0, $def->getMethodCalls());
     }
 
+    public function testProcessIgnoreInvalidArgumentInCollectionArgument()
+    {
+        $container = new ContainerBuilder();
+        $container->register('baz');
+        $def = $container
+            ->register('foo')
+            ->setArguments(array(
+                array(
+                    new Reference('bar', ContainerInterface::IGNORE_ON_INVALID_REFERENCE),
+                    $baz = new Reference('baz', ContainerInterface::IGNORE_ON_INVALID_REFERENCE),
+                    new Reference('moo', ContainerInterface::NULL_ON_INVALID_REFERENCE),
+                ),
+            ))
+        ;
+
+        $this->process($container);
+
+        $arguments = $def->getArguments();
+        $this->assertSame(array(1 => $baz, 2 => null), $arguments[0]);
+    }
+
+    public function testProcessKeepMethodCallOnInvalidArgumentInCollectionArgument()
+    {
+        $container = new ContainerBuilder();
+        $container->register('baz');
+        $def = $container
+            ->register('foo')
+            ->addMethodCall('foo', array(
+                array(
+                    new Reference('bar', ContainerInterface::IGNORE_ON_INVALID_REFERENCE),
+                    $baz = new Reference('baz', ContainerInterface::IGNORE_ON_INVALID_REFERENCE),
+                    new Reference('moo', ContainerInterface::NULL_ON_INVALID_REFERENCE),
+                ),
+            ))
+        ;
+
+        $this->process($container);
+
+        $calls = $def->getMethodCalls();
+        $this->assertCount(1, $def->getMethodCalls());
+        $this->assertSame(array(1 => $baz, 2 => null), $calls[0][1][0]);
```

Comment:
I don't have an opinion on this actually. I chose to preserve them as it avoid an extra logic for numerically indexed arrays and though it won't be harmful.

Do we have a similar behavior somewhere else that can help us to choose, or do you think it can be harmful in any way ? I'm not sure it is a good idea to rely on those indexes for anything.